lib/OpenQA/Task/Git/Clone.pm

```diff
@@ -97,7 +97,10 @@ sub _git_clone ($app, $job, $ctx, $path, $url) {
 
     return if ($requested_branch and $requested_branch !~ tr/a-f0-9//c and $git->check_sha($requested_branch));
 
-    die "NOT updating dirty git checkout at $path" if !$git->is_workdir_clean();
+    die <<~"END_OF_ERROR_MESSAGE" unless $git->is_workdir_clean;
+        NOT updating dirty Git checkout at '$path'.
+        You may disable auto-updating by setting 'git_auto_update = no' under '[scm git]' in the config.
```

Comment:
That sounds like "just always disable git_auto_update, it only makes problems" which is not what we want. How about pointing to http://open.qa/docs/#_getting_tests for details and potentially also state the bad consequence